tests/data/cifar_10/source/keras_cnn_cifar_10.py

```diff
@@ -77,122 +78,34 @@ def keras_model_fn(hyperparameters):
-
-        # Ensure that the capacity is sufficiently large to provide good random
-        # shuffling.
-        buffer_size = int(NUM_EXAMPLES_PER_EPOCH_FOR_TRAIN * 0.4) + 3 * batch_size
-        dataset = dataset.shuffle(buffer_size=buffer_size)
-
-    # Subtract off the mean and divide by the variance of the pixels.
-    dataset = dataset.map(
-        lambda image, label: (tf.image.per_image_standardization(image), label))
-    dataset.prefetch(2 * batch_size)
-
-    # Batch results by up to batch_size, and then fetch the tuple from the
-    # iterator.
-    iterator = dataset.batch(batch_size).make_one_shot_iterator()
-    images, labels = iterator.get_next()
-
-    return {PREDICT_INPUTS: images}, labels
-
-
-def _train_preprocess_fn(image, label):
-    """Preprocess a single training image of layout [height, width, depth]."""
-    # Resize the image to add four extra pixels on each side.
-    image = tf.image.resize_image_with_crop_or_pad(image, HEIGHT + 8, WIDTH + 8)
-
-    # Randomly crop a [HEIGHT, WIDTH] section of the image.
-    image = tf.random_crop(image, [HEIGHT, WIDTH, DEPTH])
-
-    # Randomly flip the image horizontally.
-    image = tf.image.random_flip_left_right(image)
-
-    return image, label
-
-
-def _dataset_parser(value):
-    """Parse a CIFAR-10 record from value."""
-    # Every record consists of a label followed by the image, with a fixed number
-    # of bytes for each.
-    label_bytes = 1
-    image_bytes = HEIGHT * WIDTH * DEPTH
-    record_bytes = label_bytes + image_bytes
-
-    # Convert from a string to a vector of uint8 that is record_bytes long.
-    raw_record = tf.decode_raw(value, tf.uint8)
-
-    # The first byte represents the label, which we convert from uint8 to int32.
-    label = tf.cast(raw_record[0], tf.int32)
-
-    # The remaining bytes after the label represent the image, which we reshape
-    # from [depth * height * width] to [depth, height, width].
-    depth_major = tf.reshape(raw_record[label_bytes:record_bytes],
-                             [DEPTH, HEIGHT, WIDTH])
-
-    # Convert from [depth, height, width] to [height, width, depth], and cast as
-    # float32.
-    image = tf.cast(tf.transpose(depth_major, [1, 2, 0]), tf.float32)
-
-    return image, tf.one_hot(label, NUM_CLASSES)
+def train_input_fn(training_dir, hyperparameters):
+    return _generate_synthetic_data(tf.estimator.ModeKeys.TRAIN, batch_size=BATCH_SIZE)
 
 
-def _record_dataset(filenames):
-    """Returns an input pipeline Dataset from `filenames`."""
-    record_bytes = HEIGHT * WIDTH * DEPTH + 1
-    return tf.data.FixedLengthRecordDataset(filenames, record_bytes)
+def eval_input_fn(training_dir, hyperparameters):
+    return _generate_synthetic_data(tf.estimator.ModeKeys.EVAL, batch_size=BATCH_SIZE)
 
 
-def _filenames(mode, data_dir):
-    """Returns a list of filenames based on 'mode'."""
-    data_dir = os.path.join(data_dir, 'cifar-10-batches-bin')
+def _generate_synthetic_data(mode, batch_size):
```

Comment:
can you put this in a different file that's used by both the TF and Keras tests?

Reply:
They are actually different. For the keras model created, the label type should match feature which is float32. Since I will talk with Marcio about whether to use real cifar10 data in tests, I think this PR is just a quick fix for current failure. We probably don't need to have a separate file with this function (and with additional parameters for type).

I will update this part after talking with Marcio in the future.